onnx_tf/backend.py

```diff
@@ -86,45 +86,68 @@ def onnx_model_to_tensorflow_rep(cls, model, strict):
     else:
       opset_import = model.opset_import
-    return cls._onnx_graph_to_tensorflow_rep(model.graph, opset_import, strict)
+    return cls._onnx_graph_to_tensorflow_rep(model.graph, opset_import, strict,
+                                             **kwargs)
 
   @classmethod
-  def _onnx_graph_to_tensorflow_rep(cls, graph_def, opset, strict):
+  def _onnx_graph_to_tensorflow_rep(cls, graph_def, opset, strict, **kwargs):
     """ Convert ONNX graph to TensorflowRep.
 
     :param graph_def: ONNX GraphProto object.
     :param opset: ONNX OperatorSetIdProto list.
     :param strict: whether to enforce semantic equivalence between the original model
       and the converted tensorflow model.
+    :kwargs: additional arguements to generate tensor_dict for model debugging
     :return: TensorflowRep object.
     """
+    # To generate tensor_dict or not, default is False
+    gen_tensor_dict = kwargs[
+        'gen_tensor_dict'] if 'gen_tensor_dict' in kwargs else False
+    # User provided input tensors, in the case the model inputs have unknown shapes
+    input_tensor_dict = kwargs[
+        'input_tensor_dict'] if 'input_tensor_dict' in kwargs else dict()
+
     handlers = cls._get_handlers(opset)
 
     # initializer: TensorProtos representing the values to initialize
     # a given tensor.
     # initialized: A list of names of the initialized tensors.
 
     if graph_def.initializer:
+      input_dict_items = cls._onnx_initializer_to_input_dict_items(
+          graph_def.initializer)
       initialized = {init.name for init in graph_def.initializer}
     else:
+      input_dict_items = []
       initialized = set()
 
     module = BackendTFModule(handlers, opset, strict, graph_def, cls)
     signatures = dict()
-
     for value_info in graph_def.input:
       if value_info.name in initialized:
         continue
       shape = list(
           d.dim_value if (d.dim_value > 0 and d.dim_param == "") else None
           for d in value_info.type.tensor_type.shape.dim)
       value_info_name = value_info.name.replace(
-            ":", "_tf_") + "_" + get_unique_suffix(
-            ) if ":" in value_info.name else value_info.name
+          ":", "_tf_") + "_" + get_unique_suffix(
+          ) if ":" in value_info.name else value_info.name
 
-      tf_spec = tf.TensorSpec(shape, data_type.onnx2tf(value_info.type.tensor_type.elem_type), value_info_name)
+      tf_spec = tf.TensorSpec(
+          shape, data_type.onnx2tf(value_info.type.tensor_type.elem_type),
+          value_info_name)
       signatures[value_info.name] = tf_spec
 
+      if gen_tensor_dict:
+        x = tf.constant(
+            0,
+            dtype=data_type.onnx2tf(value_info.type.tensor_type.elem_type),
+            name=value_info_name,
+            shape=shape
+        ) if value_info.name not in input_tensor_dict else input_tensor_dict[
+            value_info.name]
+        input_dict_items.append((value_info_name, x))
+
```

Comment:
Since we have the condition check here, it is easy and faster to set it up here. Try not to put too many unnecessary code into tf.function is the idea. I can add a check for gen_tensor_dict .